lib/nrf_modem_lib/nrf91_connectivity/Kconfig

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig NRF91_CONNECTIVITY
+	bool "nRF91 Connectivity"
+	select PDN
+	select PDN_ESM_STRERROR
+	select LTE_LINK_CONTROL
+	depends on NET_SOCKETS_OFFLOAD
+	depends on !NET_IPV6_NBR_CACHE
+	depends on !NET_IPV6_MLD
+	help
+	  Library used to provide connectivity for nRF91 based boards via Zephyr network management.
+
+if NRF91_CONNECTIVITY
+
+config NRF91_CONNECTIVITY_WORKQUEUE_STACK_SIZE
+	int "Workqueue stack size"
+	default 1024
+	help
+	  Stack size of the internal workqueue.
+
+config NRF91_CONNECTIVITY_CONNECT_TIMEOUT_SECONDS
+	int "Connect timeout in seconds"
+	default 0
+	help
+	  Time period in seconds that that the library will try to establish an LTE connection,
+	  before timing out. The connect timeout can be updated by calling
+	  conn_mgr_if_set_timeout(). Default value is 0 (timeout disabled).
+	  When the timeout is disabled, the library relies solely on the modem's internal
+	  connection handling to upkeep the connection.
+
+config NRF91_CONNECTIVITY_CONNECTION_PERSISTENCE
+	bool "Connection persistency"
+	default y
+	help
+	  Persistent connection. If the connection is lost the modem will try to reconnect
+	  it self without the need for explicit calls to the network interface.
+	  Persistence can be updated run-time by calling conn_mgr_if_set_flag() with
+	  CONN_MGR_IF_PERSISTENT as flag parameter.
+
+endif # NRF91_CONNECTIVITY
+
+module = NRF91_CONNECTIVITY
+module-str = nRF91 NET L2
```

Comment:
I'll copy here comments about the name of the module so we get a thread on it:

> > Have other names been considered other than `nrf91_connectivity`? I find it a bit misleading, since a nRF91 device can easily have "connectivity" without this library. This library integrates the nRF91 modem with the L2 interface in Zephyr, is there a name that better reflects that, for example `nrf91_net_l2`?
> 
> @lemrey The current name is just a placeholder. I agree its misleading, and misguiding for new users that might think this library could be used by itself to provide connectivity for 91 based boards. Reviewers, feel free to chime in on this topic. If not, @lemrey suggestion is the current winner. `nrf91_net_l2`

You've probably discussed this offline but I'd probably use nrf_modem_net_if, nrf_modem_net_l2, nrf_modem_net_mgmt, etc. Not sure what's the best one but current naming gives me wrong impressions about the functionality.

Do we intentionally have different naming here:
```
module = NRF91_CONNECTIVITY
module-str = nRF91 NET L2
```

Reply:
That is just residue from an older name. New name is LTE connectivity

Reply:
Clarified offline

Reply:
I agree with @trantanen, `nrf_modem_net_if` was a good suggestion, much better than `lte_connectivity` imo. @simensrostad The "connectivity" part is _very_ confusing because there is no need to have this library to have "LTE connectivity".

As I user, looking at this library I would think I need this to have LTE connectivity and that's 100% incorrect.
I think this library should have a name that reflects the fact that it wraps around Zephyr's networking API abstractions.